Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -9,12 +9,9 @@ namespace FluentAssertions.Specialized
     /// Contains a number of methods to assert that an asynchronous method yields the expected result.
     /// </summary>
     [DebuggerNonUserCode]
-    public class AsyncFunctionAssertions : ActionAssertions
+    public class AsyncFunctionAssertions : DelegateAssertions<Func<Task>>
     {
-        public AsyncFunctionAssertions(Func<Task> subject, IExtractExceptions extractor) : base(() =>
-        {
-            subject().GetAwaiter().GetResult();
-        }, extractor)
+        public AsyncFunctionAssertions(Func<Task> subject, IExtractExceptions extractor) : base(subject, extractor)
         {
             Subject = subject;
```

Comment:
🙃 Redundant assignment?

Reply:
Not redundant, as this initialises the new property which hides the base definition on line 22. 
This means we have a value which is used for populating both a Func<Task> and Delegate property.